### ISSUE TITLE
UNIVERSAL NOSLOW BYPASS

### ISSUE THREAD

**Issue by C00LC0D35** [GrimAnticheat/Grim#1423]
### Describe the bypass and how to replicate it

Basically, to bypass noslow, a client can just send two c09 (switchslot) packets to sword and back. This will bypass all noslow checks for sword, food, etc. This can easily be blocked by checking badpackets (two c09 in one tick = flag).

### Grim version

Latest

### Server version

Any (tested on 1.8)

### Plugins

No

**Comment by ManInMyVan** [comment]
Duplicate of #1091

> This can easily be blocked by checking badpackets (two c09 in one tick = flag).

This was what BadPacketM did, it was removed in 08d3392b0f0ed21b23aeb7cf35e683e50a7d7166 for not being a valid check.

**Comment by z7087** [comment]
vanilla client can send 3 c09 in a tick (in my testing)
https://github.com/z7087/Grim/tree/c09

**Comment by ManInMyVan** [comment]
> vanilla client can send 3 c09 in a tick (in my testing)

If you're using an item, does it keep using the item if you change slots and back in one tick (like the noslow)? if not, can we reset item usage (should make it only swords)?

**Comment by z7087** [comment]
> If you're using an item, does it keep using the item if you change slots and back in one tick (like the noslow)?

i didnt find how client stop item using so likely yes (in 1.12.2, client and server)
